generator/internal/scaffold_generator.cc

```diff
@@ -136,6 +136,12 @@ std::map<std::string, std::string> ScaffoldVars(
   auto const library = LibraryName(service.product_path());
   vars["copyright_year"] = service.initial_copyright_year();
   vars["library"] = library;
+  vars["product_namespace"] = absl::StrReplaceAll(
+      absl::StripSuffix(
+          absl::StrCat(library, "_",
+                       ServiceSubdirectory(service.product_path())),
+          "/"),
+      {{"/", "::"}});
```

Comment:
shouldn't `::` be `_` ?

So like: `foo/bar/baz/v1/` would become `foo_bar_baz_v1`.

Reply:
Yes.  I failed to test nested services.  Let me fix that.  Thanks.